src/Processors/Executors/PipelineExecutor.cpp

```diff
@@ -744,9 +739,7 @@ void PipelineExecutor::executeImpl(size_t num_threads)
             });
         }
 
-        for (auto & thread : threads)
-            if (thread.joinable())
-                thread.join();
+        threads.wait();
```

Comment:
Is this the part which this comment applies to? https://github.com/ClickHouse/ClickHouse/pull/15177#issuecomment-697857284

Can you put this comment into the code as well?

Reply:
>Is this the part which this comment applies to? #15177 (comment)

Yep

>Can you put this comment into the code as well?


Ok (there is description in the commit, but it worth adding a comment into the code agree)